sdk_contrib/aws_v1_support/src/main/java/io/opentelemetry/sdk/contrib/trace/aws/propagation/AWSXRayPropagator.java

```diff
@@ -0,0 +1,238 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opentelemetry.sdk.contrib.trace.aws.propagation;
+
+import static io.opentelemetry.internal.Utils.checkNotNull;
+
+import io.grpc.Context;
+import io.opentelemetry.context.propagation.HttpTextFormat;
+import io.opentelemetry.internal.StringUtils;
+import io.opentelemetry.trace.DefaultSpan;
+import io.opentelemetry.trace.Span;
+import io.opentelemetry.trace.SpanContext;
+import io.opentelemetry.trace.SpanId;
+import io.opentelemetry.trace.TraceFlags;
+import io.opentelemetry.trace.TraceId;
+import io.opentelemetry.trace.TraceState;
+import io.opentelemetry.trace.TracingContextUtils;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import javax.annotation.Nullable;
+
+/**
+ * Implementation of the AWS X-Ray Trace Header propagation protocol. See <pre><a href=
+ * https://https://docs.aws.amazon.com/xray/latest/devguide/xray-concepts.html#xray-con
+ * cepts-tracingheader>AWS Tracing header spec</a></pre>.
+ *
+ * To register the X-Ray propagator together with default propagator:
+ * <pre>
+ * OpenTelemetry.setPropagators(
+ *   DefaultContextPropagators.builder()
+ *     .addHttpTextFormat(new HttpTraceContext())
+ *     .addHttpTextFormat(new AWSXRayPropagator()).build());
+ * </pre>
+ */
+public class AWSXRayPropagator implements HttpTextFormat {
+
+  private static final Logger logger = Logger.getLogger(AWSXRayPropagator.class.getName());
+
+  static final String TRACE_HEADER_KEY = "X-Amzn-Trace-Id";
+  private static final char TRACE_HEADER_DELIMITER = ';';
+  private static final char KV_DELIMITER = '=';
+
+  private static final String TRACE_ID_KEY = "Root";
+  private static final String TRACE_ID_VERSION = "1";
+  private static final char TRACE_ID_DELIMITER = '-';
+  private static final int TRACE_ID_FIRST_PART_LENGTH = 8;
+  private static final int TRACE_ID_SECOND_PART_LENGTH = 24;
+
+  private static final String PARENT_ID_KEY = "Parent";
+  private static final int PARENT_ID_LENGTH = 16;
+
+  private static final String SAMPLED_FLAG_KEY = "Sampled";
+  private static final int SAMPLED_FLAG_LENGTH = 1;
+  private static final char IS_SAMPLED = '1';
+  private static final char NOT_SAMPLED = '0';
+
+  private static final List<String> FIELDS = Collections.singletonList(TRACE_HEADER_KEY);
+
+  @Override
+  public List<String> fields() {
+    return FIELDS;
+  }
+
+  @Override
+  public <C> void inject(Context context, @Nullable C carrier, Setter<C> setter) {
+    Objects.requireNonNull(context, "context");
+    Objects.requireNonNull(setter, "setter");
+
+    Span span = TracingContextUtils.getSpanWithoutDefault(context);
```

Comment:
Think it's better to use `.getSpan(context)` and check `span.getContext().isValid()`. From what I can tell, this current method may return an invalid span too, so either we check both null and valid or just valid.

Reply:
will check valid to align with the [default propagator](https://github.com/open-telemetry/opentelemetry-java/blob/master/api/src/main/java/io/opentelemetry/trace/propagation/HttpTraceContext.java#L83).